### PR TITLE
Feature/cg 488 Role service

### PR DESCRIPTION
**Pull request by tyler-dunkel**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,17 +34,19 @@ Authenticate the CloudGraph k8s Provider any of the following ways:
...
+| ingress                      | namespace          |
+| namespace                    | ALL SERVICES       |
+| node                         | namespace          |
+| persistentVolume             | namespace          |
+| persistentVolumeClaim        | namespace          |
+| pod                          | namespace          |
+| secret                       | namespace          |
+| service                      | namespace          |
+| serviceAccount               | namespace          |
+| storageClass                 | namespace          |
```
Comment (csbrandt) [inline README.md]: is this missing role?

File: src/services/role/format.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    apiGroups,
+    nonResourceUrls: nonResourceURLs,
+    resources,
+    resourceNames,
+    verbs
+  })) ?? []
+
+  return {
+    id: formattedMetadata.id,
+    apiVersion,
```
Comment (csbrandt) [inline src/services/role/format.ts:33]: I dont think this is related to this pr specifically but it looks like `apiVersion` and `kind` is null across all services

File: src/services/role/format.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    kind,
+    metadata,
+    rules
+  } = entity
+
+  const formattedMetadata = formatMetadata(metadata)
+
+  const mappedRules = rules?.map(({ apiGroups, nonResourceURLs, resources, resourceNames, verbs }) => ({
+    id: cuid(),
+    apiGroups,
```
Comment (csbrandt) [inline src/services/role/format.ts:24]: I guess `apiGroups` comes across like this but its a bit weird to see in the output

```
            "apiGroups": [
              ""
            ],
```

File: src/services/role/format.ts (hunk 2 above)
Comment (tyler-dunkel) [reply src/services/role/format.ts:33]: yea I dont know why none of the services in our sandbox have them filled in but they are a part of the obj definition. I can ask Chris K.

File: src/services/role/format.ts (hunk 3 above)
Comment (tyler-dunkel) [reply src/services/role/format.ts:24]: yea Idk if there is much I can do there. idk why it is setup that way. I suppose I could filter out empty strings from the array but maybe there is a reason for it.

File: README.md (hunk 1 above)
Comment (tyler-dunkel) [reply README.md]: might of missed that one, ill add it in

**Comment by csbrandt** [comment]
approve. where is the review changes button?

**Comment by tyler-dunkel** [comment]
> approve. where is the review changes button?

good question lol idk

**Pull request merged**